docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,212 @@
+
+Namespace ID size (bytes) | Unencoded NMT proof size (bytes) | Protobuf encoded NMT proof size (bytes) | Protobuf encoded NMT proof with [gzip](https://pkg.go.dev/compress/gzip) (bytes)
+--------------------------|----------------------------------|-----------------------------------------|---------------------------------------------------------------------------------
+8                         | 336                              | 354                                     | 382
+16                        | 448                              | 466                                     | 408
+20                        | 504                              | 522                                     | 466
+32                        | 672                              | 690                                     | 630
+
+### Blob Inclusion Proof Size
+
+Blob inclusion proofs haven't yet been implemented so this proposal can't precisely determine the impact on blob inclusion proofs. A naive implementation of blob inclusion proofs may return NMT proofs for all shares that a blob occupies, in other words one NMT proof per row that a blob spans. Assuming shares are 512 bytes, square size is 128, and a blob is less than 128 shares, a blob would occupy a maximum of 2 rows. Therefore, the namespace ID size's impact on blob inclusion proofs would be approximately 2 * the impact on NMT proofs. A [blob size independent inclusion proof](https://github.com/celestiaorg/celestia-app/blob/6d27b78aa64a749a808e84ea682352b8b551fbd7/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md?plain=1#L19) is likely smaller than this naive implementation because it depends on the number of shares that a PFB transaction spans (likely significantly fewer than 2 rows).
+
+### Share Size
+
+Another tradeoff to consider is the size of the namespace ID in the share. Since a share is a fixed 512 bytes, a share's capacity for blob data decreases as the namespace ID increases.
+
+| Namespace ID size (bytes) | Namespace ID size (bytes) / 512 (bytes) |
+|---------------------------|-----------------------------------------|
+| 8                         | 1.5%                                    |
+| 16                        | 3.1%                                    |
+| 20                        | 3.9%                                    |
+| 32                        | 6.2%                                    |
+
+## Open Questions
+
+1. What are the performance implications on celestia-node for a larger namespace ID size?
+1. Is it possible to mitigate some tradeoffs when adopting a large namespace ID size?
+    1. It may be possible to decrease the bandwidth requirements for NMT proofs by using lossless compression (proposed by @evan-forbes) and explored above.
+    1. It may be possible to avoid writing the namespace ID to continuation blob shares (proposed by @nashqueue)
```

Comment:
[Question] I'm uncertain how we could incorporate this idea into NMT, since the block shares' corresponding leaves in NMT must have distinct namespaces. Consequently, we may not be able to apply this optimization in the final NMT representation.

Reply:
Yea, this optimization would happen before writing to the leaf. There's a Slack thread exploring this topic here: https://celestia-team.slack.com/archives/C02PFFCGMNW/p1677197644501209

The rough idea is that shares wouldn't contain the namespace ID but the entire data square (or at least preceding rows) would need to be reconstructed prior to constructing the NMT so that leaves pushed to the NMT would contain the namespace ID.